astropy/constants/__init__.py

```diff
@@ -23,12 +24,42 @@
 except ImportError:
     pass
 
-from .constant import Constant, EMConstant  # noqa
-from . import si  # noqa
-from . import cgs  # noqa
-from . import codata2014, iau2015  # noqa
+import astropy.config as _config
+
 from . import utils as _utils
 
+
+class Conf(_config.ConfigNamespace):
+    """
+    Configuration parameters for `astropy.constants`.
+    """
+    # NOTE: Update this when default changes.
+    physical_constants = _config.ConfigItem(
+        'codata2014',
+        'Version of physical constants to use.')
+    astronomical_constants = _config.ConfigItem(
+        'iau2015',
+        'Version of astronomical constants to use.')
+
+
+conf = Conf()
+
+if conf.physical_constants == 'codata2014':
+    from . import codata2014 as codata
+elif conf.physical_constants == 'codata2010':
+    from .astropyconst13 import codata2010 as codata
+else:
+    raise ValueError('Invalid configuration for physical constants: {}'
+                     .format(conf.physical_constants))
+
+if conf.astronomical_constants == 'iau2015':
+    from . import iau2015 as iaudata
+elif conf.physical_constants == 'codata2010':
+    from .astropyconst13 import iau2012 as iaudata
+else:
+    raise ValueError('Invalid configuration for physical constants: {}'
```

Comment:
Copy & paste error: should be "astronomical" here, not physical (also in the `.format`).

Reply:
Doh! Good catch! In the config test, I wasn't checking the prior version of astronomical constants, which I think would have caught this error. I've added a check.